### ISSUE TITLE
[java] ImmutableField false positive for Java entities

### ISSUE THREAD

**Issue by road723** [pmd/pmd#3888]
**Affects PMD Version:** 6.44

**Rule:** [ImmutableField](https://pmd.github.io/latest/pmd_rules_java_design.html#immutablefield)

**Description:** Per Java EE spec (https://docs.oracle.com/javaee/6/tutorial/doc/bnbqa.html) Java entity classes must not be declared final. Likewise no methods or persistent instance variables must be declared final. The last part is especially true for this rule as now with 6.44 and the rule change to ImmutableField we are seeing false positives on entities since setters are not always required for private variables. In addition, these entities don't require annotations to work correctly so the ignoredAnnotations property does not help either.

**Code Sample demonstrating the issue:**

```java
@Entity
@Table(name = "Users")
public class UserWriteEntity implements Serializable {

private boolean xWrite;

public boolean isXWrite() {
		return xWrite;
	}

}
```

**Expected outcome:** 

PMD reports a violation for private boolean xWrite but according to the Java EE spec this is correct and it should not be made final.

**Running PMD through:** Ant

**Comment by road723** [comment]
Probably related but this also points out that simply excluding annotations will not resolve

**Comment by Thunderforge** [comment]
This also affects mocked objects via Mockito.

```java
@ExtendWith(MockitoExtension.class)
class FooServiceTest {
    @InjectMocks
    private FooService fooService;

    @Mock
    private BarService barService;

    @Test
    void fooServiceCallsBarService() {
        fooService.doFoo();

        verify(barService)
            .doBar();
}
```

PMD complains that the annotated objects should be `final`, which isn't possible.
